action/protocol/staking/read_state.go

```diff
@@ -184,3 +198,28 @@ func getPageOfCandidates(candidates CandidateList, offset, limit int) CandidateL
 	}
 	return res
 }
+
+func getTotalStakedAmount(ctx context.Context, sr protocol.StateReader) (uint64, string) {
+	csr, err := ConstructBaseView(sr)
+	if err != nil {
+		return 0, "0"
+	}
+
+	chainCtx := protocol.MustGetBlockchainCtx(ctx)
+	hu := config.NewHeightUpgrade(&chainCtx.Genesis)
+	switch {
+	case hu.IsPost(config.Greenland, csr.Height()):
+		// after Greenland, read state from db
+		var total totalAmount
+		height, err := sr.State(&total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey))
+		if err != nil {
+			return height, "0"
+		}
+		return height, total.amount.String()
+	case hu.IsPost(config.Fairbank, csr.Height()):
+		// fairbank to greenland, read from bucket pool
+		return csr.Height(), csr.BucketPool().Total().String()
+	default:
+		return csr.Height(), "0"
```

Comment:
isn't these two the same? if before fairbank, the total amount should be always 0?

Reply:
fixed